packages/core/src/credential/Credential.ts

```diff
@@ -292,118 +277,114 @@ export function fromClaim(
   return credential
 }
 
-export interface TrustPolicy {
-  /**
-   * If set to true, attestations issued directly by this identity are accepted.
-   */
-  isAttester?: boolean
-  /**
-   * If set to true, attestations linked to a delegation node which this identity has control over are accepted.
-   */
-  isDelegator?: boolean
+type VerifyOptions = {
+  ctype?: ICType
+  challenge?: string
+  didResolveKey?: DidResolveKey
+  throwOnRevoked?: boolean
+}
+
+/**
+ * Verifies data structure & data integrity of a credential object.
+ * This combines all offline sanity checks that can be performed on an ICredential object.
+ * A credential is valid only iff it is well formed and there is an onchain attestation record referencing its root hash.
+ * To check the latter condition as well, you need to call [[verifyCredential]] or [[verifyPresentation]].
+ *
+ * @param credential - The object to check.
+ * @param options - Additional parameter for more verification steps.
+ * @param options.ctype - CType which the included claim should be checked against.
+ */
+export function verifyWellFormed(
+  credential: ICredential,
+  { ctype }: VerifyOptions = {}
+): void {
+  verifyDataStructure(credential)
+  verifyDataIntegrity(credential)
+
+  if (ctype) {
+    verifyClaimAgainstSchema(credential.claim.contents, ctype)
+  }
 }
-export type TrustPolicies = Record<DidUri, TrustPolicy>
 
 /**
  * Queries the attestation record for a credential and matches their data. Fails if no attestation exists, if it is revoked, or if the attester is unknown.
  *
  * @param credential The [[ICredential]] whose attestation status should be checked.
- * @param allowedAuthorities A map from DIDs to trust policies to be applied for that DID. If the credential's attestation cannot be linked to one of the identies in this set, verifiation will fail. If omitted, the issuer of the attestation will not be checked.
- * @param allowRevoked If true, a revoked attestation will not fail verification.
+ * @param throwOnRevoked If not set to false, a revoked attestation will result in a rejected Promise.
  * @returns Information on the attester and revocation status of the on-chain attestation, as well as info on which trust policy led to acceptance of the credential.
  */
 export async function verifyAttested(
   credential: ICredential,
-  allowedAuthorities?: TrustPolicies,
-  allowRevoked = false
+  throwOnRevoked = true
```

Comment:
Why doesn't this function also returns a `VerifiedCredential`? Plus, I agree that removing the `throwOnRevoked` flag is more future-proof and less prone to mistakes, as people might want to check other stuff as well.

Reply:
Because it does not actually verify a credential. Just checks the attestation. Name suggestions for that function are welcome.